jmdaemon/jmdaemon/onionmc.py

```diff
@@ -1500,14 +1500,24 @@ def send_peers(self, requesting_peer: OnionPeer,
                     # if a connection has no reachable destination,
                     # don't forward it
                     continue
-                peerlist.add(p.get_nick_peerlocation_ser())
+                try:
+                    peerlist.add(p.get_nick_peerlocation_ser())
+                except OnionPeerError:
+                    pass
         else:
             # since the peer may already be removed from self.peers,
             # we don't limit except by filter:
             for p in peer_filter:
-                peerlist.add(p.get_nick_peerlocation_ser() + NICK_PEERLOCATOR_SEPARATOR + "D")
+                try:
+                    peerlist.add(p.get_nick_peerlocation_ser(
+                        ) + NICK_PEERLOCATOR_SEPARATOR + "D")
+                except OnionPeerError:
+                    pass
         # For testing: dns won't usually participate:
-        peerlist.add(self.self_as_peer.get_nick_peerlocation_ser())
+        try:
+            peerlist.add(self.self_as_peer.get_nick_peerlocation_ser())

```

Comment:
Yeah, in the non-disconnect block, I thought 'may as well just wrap all of them'. For the self_as_peer case I realised it didn't make too much sense, but I actually didn't notice that the other one was preceded by checking if nick == "".

So that was sloppy, I agree that both of these should be removed, a case of that error occurring means an actual code bug. Will change.